NYPL-discovery/discovery-front-end
# SCC-3206: Remove Design Toolkit

**What's this do?**

- This removes the `@nypl/design-toolkit` npm package.
- Many styles were dependent on the Design Toolkit so some styles were copied over to fix the broken UI styles. This was done throughout the app for all the pages that needed this.
- As we move forward with the Design System, the added SCSS in this PR will be removed.
- Some of SCSS variables are commented out and not removed. This is just to keep a record of the name/color value until it is moved over to the Reservoir Design System colors (out of scope for this PR).
- The biggest difference from this changes is the checkboxes. This should be an okay change since this moves closer to the styles from the Design System. Since this PR doesn't fix/update the checkbox specifically, it will look odd until those checkboxes are updated. A clear example of this can be found in the filter dropdown. The older checkboxes were green when selected but now they are the browser default blue:

<img width="1411" alt="Screen Shot 2022-08-11 at 11 28 41 AM" src="https://user-images.githubusercontent.com/1280564/184209378-3aadd79c-d446-4f5e-9a18-07643b3bd4ae.png">

**Why are we doing this? (w/ JIRA link if applicable)**

- We are doing this because the Design Toolkit is outdated and no longer maintained. The Reservoir Design System has been implemented removing this package will help move closer to the NYPL branded styles.
- This also helps decrease the amount of security issues found in npm package dependencies. This is a rough view of how many vulnerabilities were fixed by removing this package

<img width="632" alt="Screen Shot 2022-08-11 at 2 11 45 PM" src="https://user-images.githubusercontent.com/1280564/184209813-ce355885-1eda-432e-b0e5-94144b542020.png">


**Do these changes have automated tests?**

This is a UI change so tests were not impacted except for the `ResultsCount.test.jsx` file were the `shallow` function was replaced with the `mount` function.

**How should this be QAed?**

Go through every page to make sure it looks right and just like the production view. This includes all SHEP pages, account pages (though they were not really impacted by this change),  and EDD pages.

**Dependencies for merging? Releasing to production?**
N/A

**Has the application documentation been updated for these changes?**

**Did someone actually run this code to verify it works?**
